wnfs/benches/namefilter.rs

```diff
@@ -0,0 +1,74 @@
+use criterion::{black_box, criterion_group, criterion_main, Criterion, Throughput};
+use wnfs::{dagcbor, private::namefilter::Namefilter};
+
+fn add(c: &mut Criterion) {
+    c.bench_function("namefilter add", |b| {
+        b.iter(|| {
+            let mut namefilter = black_box(Namefilter::default());
+            for i in 0..50 {
+                black_box(namefilter.add(&i.to_string()));
+            }
+        })
+    });
+}
+
+fn contains(c: &mut Criterion) {
+    let mut namefilter = Namefilter::default();
+    for i in 0..50 {
+        namefilter.add(&i.to_string());
+    }
+
+    c.bench_function("namefilter contains", |b| {
+        b.iter(|| {
+            for i in 0..50 {
+                assert!(namefilter.contains(&i.to_string()));
+            }
+        })
+    });
+}
+
+fn saturate(c: &mut Criterion) {
+    let mut namefilter = Namefilter::default();
+    for i in 0..50 {
+        namefilter.add(&i.to_string());
+    }
+
+    c.bench_function("namefilter saturate", |b| {
+        b.iter(|| {
+            black_box(namefilter.saturate());
```

Comment:
I'm not sure whether this is actually doing anything for two reasons:
- I worry that `namefilter.saturate()` is detected to just be a constant value and thus optimized. (Do we want to put a `black_box(namefilter).saturate()` here instead?)
- Since that's a 2048-bit bloom filter with 30 hashes & we're filling it over the half-way line with 50 elements, it actually is already over-saturated and saturation will just detect that and bail immediately. This namefilter should be filled with some random element and we should test the worst case performance of saturating an almost empty namefilter.

Reply:
Oops. Totally missed that.